token-metadata/program/src/processor/metadata/print.rs

```diff
@@ -0,0 +1,194 @@
+        MintNewEditionFromMasterEditionViaTokenLogicArgs,
+    },
+};
+
+pub fn print<'a>(
+    program_id: &'a Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    args: PrintArgs,
+) -> ProgramResult {
+    let context = Print::to_context(accounts)?;
+
+    match args {
+        PrintArgs::V1 { .. } => print_v1(program_id, context, args),
+    }
+}
+
+fn print_v1(_program_id: &Pubkey, ctx: Context<Print>, args: PrintArgs) -> ProgramResult {
+    // Get the args for the instruction
+    let PrintArgs::V1 { edition } = args;
+    // let account_info_iter = &mut accounts.iter();
+
+    // let new_metadata_account_info = next_account_info(account_info_iter)?;
+    // let new_edition_account_info = next_account_info(account_info_iter)?;
+    // let master_edition_account_info = next_account_info(account_info_iter)?;
+    // let mint_info = next_account_info(account_info_iter)?;
+    // let token_record_info = next_account_info(account_info_iter)?;
+    // let edition_marker_info = next_account_info(account_info_iter)?;
+    // let mint_authority_info = next_account_info(account_info_iter)?;
+    // let payer_account_info = next_account_info(account_info_iter)?;
+    // let owner_account_info = next_account_info(account_info_iter)?;
+    // let token_account_info = next_account_info(account_info_iter)?;
+    // let update_authority_info = next_account_info(account_info_iter)?;
+    // let master_metadata_account_info = next_account_info(account_info_iter)?;
+    // let token_program_account_info = next_account_info(account_info_iter)?;
+    // let system_account_info = next_account_info(account_info_iter)?;
+
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_metadata_info = ctx.accounts.edition_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_account_info = ctx.accounts.edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_info = ctx.accounts.edition_mint_info;
+    let edition_token_account_owner_info = ctx.accounts.edition_token_account_owner_info;
+    let edition_token_account_info = ctx.accounts.edition_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_authority_info = ctx.accounts.edition_mint_authority_info;
+    let edition_token_record_info = ctx.accounts.edition_token_record_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_edition_info = ctx.accounts.master_edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_marker_pda_info = ctx.accounts.edition_marker_pda_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let payer_info = ctx.accounts.payer_info;
```

Comment:
Should there be a check for payer is signer in `process_mint_new_edition_from_master_edition_via_token_logic`?  I think it would probably fail when it uses the payer, but it might be more helpful for there to be an explicit payer is signer check earlier.

Reply:
Good idea!